infra/hasura/metadata/deploy-tenant.sh

```
@@ -0,0 +1,269 @@
+    local admin_secret="$3"
+    
+    # Create temporary directory
+    local temp_dir
+    temp_dir=$(mktemp -d)
+    
+    # Create metadata source and get tenant name
+    local tenant_name
+    tenant_name=$(create_metadata_source "$tenant" "$temp_dir" "$hasura_endpoint" "$admin_secret")
+    
+    # Check if source creation was successful
+    if [ $? -ne 0 ]; then
```

Shellcheck (0.11.0)

[style] Check exit code directly with e.g. 'if ! mycmd;', not indirectly with $?.

(SC2181)